en/asgardeo/docs/apis/restapis/actions.yaml

```diff
@@ -0,0 +1,859 @@
+          source: |
+            curl --location --request PUT 'https://api.asgardeo.io/t/{organization-name}/api/server/v1/actions/{actionType}/{actionId}/apiKey' \
+            -H 'Content-Type: application/json' \
+            -H 'Accept: application/json' \
+            -H 'Authorization: Bearer {bearer_token}' \
+            -d '{
+              "properties": {
+                "header": "header",
+                "value": "value"
+              }
+            }'
+        - lang: Curl(None)
+          source: |
+            curl --location --request PUT 'https://api.asgardeo.io/t/{organization-name}/api/server/v1/actions/{actionType}/{actionId}/none' \
+            -H 'Content-Type: application/json' \
+            -H 'Accept: application/json' \
+            -H 'Authorization: Bearer {bearer_token}' \
+            -d '{
+              "properties": {}
+            }'
+      x-codegen-request-body-name: body
+
+components:
+  securitySchemes:
+    OAuth2:
+      type: oauth2
+      flows:
+        authorizationCode:
+          authorizationUrl: 'https://api.asgardeo.io/t/{org-name}/oauth2/authorize'
+          tokenUrl: 'https://api.asgardeo.io/t/{org-name}/oauth2/token'
+          scopes: 
+            read: internal_action_mgt_view
+            write: internal_action_mgt_create
+            update: internal_action_mgt_update
+            delete: internal_action_mgt_delete
+
+  schemas:
+    ActionModel:
+      type: object
+      required:
+        - name
+        - endpoint
+      properties:
+        name:
+          type: string
+          example: Access Token Pre Issue
+          minLength: 1
+          maxLength: 255
+        description:
+          type: string
+          example: This is the configuration of pre-action for issuing access token.
+          maxLength: 255
+        endpoint:
+          $ref: '#/components/schemas/Endpoint'
+
+    Endpoint:
+      type: object
+      required:
+        - uri
+        - authentication
+      properties:
+        uri:
+          type: string
+          example: https://myextension.com/token
+          pattern: '^https?://.+'
+        authentication:
+          $ref: '#/components/schemas/Authentication'
+
+    EndpointResponse:
+      type: object
+      required:
+        - uri
+        - authentication
+      properties:
+        uri:
+          type: string
+          example: https://myextension.com/token
+          pattern: '^https?://.+'
+        authentication:
+          $ref: '#/components/schemas/AuthenticationTypeResponse'
+
+    ActionResponse:
+      type: object
+      properties:
+        id:
+          type: string
+          example: '24f64d17-9824-4e28-8413-de45728d8e84'
+        type:
+          type: string
+          enum:
+            - PRE_ISSUE_ACCESS_TOKEN
+            - PRE_UPDATE_PASSWORD
+            - PRE_UPDATE_PROFILE
+            - PRE_REGISTRATION
+        name:
+          type: string
+          example: Access Token Pre Issue
```

Comment:
This example should be using the defined enum isn't it ?

Reply:
This is the action name. Only the action type comes with the enum